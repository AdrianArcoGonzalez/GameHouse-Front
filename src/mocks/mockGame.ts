import { Game } from "../interfaces/interfaces";

export const mockGame: Game = {
  category: "MOBA",
  company: "Riot",
  dislikes: 40,
  likes: 100,
  image: "images/leagueoflegends.jpg",
  owner: "Admin",
  sinopsis:
    "Lorem ipsum dolor, sit amet consectetur adipisicing elit. Aliquam odio ut odit, illo sit eligendi nobis explicabo eos, aliquid qui inventore provident minus impedit consequuntur asperiores iure recusandae culpa quam?",

  title: "League of Legends",
  reviews: [""],
};

export const mockGame2: Game = {
  category: "Adventure",
  company: "Bethesda",
  dislikes: 40,
  likes: 100,
  image: "images/thelastofus.webp",
  owner: "Admin",
  sinopsis:
    "Lorem ipsum dolor, sit amet consectetur adipisicing elit. Aliquam odio ut odit, illo sit eligendi nobis explicabo eos, aliquid qui inventore provident minus impedit consequuntur asperiores iure recusandae culpa quam?",

  title: "The Last of Us",
  reviews: [""],
};
export const mockGame3: Game = {
  category: "Shooter",
  company: "Bethesda",
  dislikes: 40,
  likes: 100,
  image: "images/tarkov.jpg",
  owner: "Admin",
  sinopsis:
    "Lorem ipsum dolor, sit amet consectetur adipisicing elit. Aliquam odio ut odit, illo sit eligendi nobis explicabo eos, aliquid qui inventore provident minus impedit consequuntur asperiores iure recusandae culpa quam?",

  title: "Escape from Tarkov",
  reviews: [""],
};

export const mockGameArray: Game[] = [
  mockGame,
  mockGame2,
  mockGame3,
  mockGame,
  mockGame2,
  mockGame3,
];
