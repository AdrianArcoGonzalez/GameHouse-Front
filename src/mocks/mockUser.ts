const mockUser = {
  username: "admin",
  isLogged: true,
  token: "123123123",
  id: "123123123",
};
export default mockUser;
