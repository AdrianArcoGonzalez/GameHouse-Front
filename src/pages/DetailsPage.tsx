import { useEffect, useState } from "react";
import { useParams } from "react-router-dom";
import GameDetails from "../components/GameDetails/GameDetails";
import heroTexts from "../components/HeroSection/heroData/heroData";
import HeroSection from "../components/HeroSection/HeroSection";
import useGamesApi from "../hooks/useGamesApi";
import { Game } from "../interfaces/interfaces";

const initialState: Game = {
  category: "",
  company: "",
  dislikes: 0,
  id: "",
  image: "",
  likes: 0,
  owner: "",
  sinopsis: "",
  title: "",
  reviews: [""],
};

const DetailsPage = (): JSX.Element => {
  const [game, setGame] = useState(initialState);
  const { getOneGameById } = useGamesApi();
  const { id } = useParams();

  useEffect(() => {
    (async () => {
      const game = await getOneGameById(id!);
      setGame(game);
    })();
  }, [getOneGameById, id]);

  return (
    <>
      <HeroSection
        text={heroTexts.details}
        srcImage={"/images/detailsHeader.jpg"}
      />
      <GameDetails game={game} />
    </>
  );
};

export default DetailsPage;
