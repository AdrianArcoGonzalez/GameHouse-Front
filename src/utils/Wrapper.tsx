import * as reactRedux from "react-redux";
import { store } from "../store/store";

export interface WrapperProps {
  children: JSX.Element | JSX.Element[];
}

export const Wrapper = ({ children }: WrapperProps): JSX.Element => {
  return <reactRedux.Provider store={store}>{children}</reactRedux.Provider>;
};
