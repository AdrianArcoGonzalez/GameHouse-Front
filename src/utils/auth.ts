import jwt from "jwt-decode";

const decodeToken = (token: string) => {
  const payloadToken: {
    id: string;
    username: string;
    iat: number;
  } = jwt(token);

  const user = {
    token: token,
    id: payloadToken.id,
    username: payloadToken.username,
    isLogged: true,
  };
  return user;
};

export default decodeToken;
