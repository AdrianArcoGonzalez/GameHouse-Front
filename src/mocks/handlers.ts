import { rest } from "msw";
import { mockGame, mockGameArray } from "./mockGame";

const url = process.env.REACT_APP_URL_BACK;

export const handlers = [
  rest.post(`${url}games/users/login`, async (req, res, ctx) => {
    const body = await req.json();
    if (!body.username || !body.password) {
      return res(
        ctx.status(403),
        ctx.json({
          error: "Wrong data",
        })
      );
    }

    return res(
      ctx.status(200),
      ctx.json({
        user: {
          token:
            "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyTmFtZSI6InZ2dnZ2IiwiaWQiOiI2MzBmMjkwOTJmMDU2MDMzMDFjOTY2ZjUiLCJpYXQiOjE2NjE5Mzk2MTl9.Zff5fV2EnVvFNBKZSNbmlE-fcZQbBoMhbq-q0KTjH7I",
        },
      })
    );
  }),

  rest.post(`${url}games/users/register`, async (req, res, ctx) => {
    const request: any = req;
    const userFormData: string = await request._body.get("user");
    const user = JSON.parse(userFormData);
    if (user.password === "") {
      return res(
        ctx.status(400),
        ctx.json({
          error: "Wrong data",
        })
      );
    }
    return res(
      ctx.status(200),
      ctx.json({
        message: "User created",
      })
    );
  }),

  rest.get(`${url}games/games`, async (_req, res, ctx) => {
    return res(ctx.status(200), ctx.json({ games: mockGameArray }));
  }),

  rest.get(`${url}games/games/46546549874123`, (_req, res, ctx) => {
    return res(ctx.status(200), ctx.json({ game: mockGame }));
  }),

  rest.get(`${url}games/games/123123123123123123`, (_req, res, ctx) => {
    return res(ctx.status(404), ctx.json(new Error()));
  }),

  rest.delete(`${url}games/games/`, async (_req, res, ctx) => {
    return res(ctx.status(200), ctx.json({ game: mockGame }));
  }),
];
