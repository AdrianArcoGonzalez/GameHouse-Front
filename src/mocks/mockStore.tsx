import { configureStore, createReducer } from "@reduxjs/toolkit";
import { Game, User } from "../interfaces/interfaces";
import { mockGameArray } from "./mockGame";

const initialUserState = {
  username: "admin",
  id: "123123123",
  isLogged: true,
  token: "123123123.123123123.1231231",
};

const initialGameState: Game[] = mockGameArray;

const userInitialState = { ...initialUserState, isLogged: true };
const gameInitialState = { ...initialGameState, mockGameArray };

const mockUserReducer = createReducer<User>(userInitialState, (builder) => {
  builder.addDefaultCase((state: User) => state);
});
const mockGameReducer = createReducer<Game[]>(gameInitialState, (builder) => {
  builder.addDefaultCase((state: Game[]) => state);
});

const mockStore = configureStore({
  reducer: {
    user: mockUserReducer,
    games: mockGameReducer,
  },
});

export default mockStore;
