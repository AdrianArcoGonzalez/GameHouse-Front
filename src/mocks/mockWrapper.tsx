import * as reactRedux from "react-redux";
import mockStore from "./mockStore";

export interface MockWrapperProps {
  children: JSX.Element | JSX.Element[];
}

export const mockWrapper = ({ children }: MockWrapperProps): JSX.Element => {
  return (
    <reactRedux.Provider store={mockStore}>{children}</reactRedux.Provider>
  );
};
