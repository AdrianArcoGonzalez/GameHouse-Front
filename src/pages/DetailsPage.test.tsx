import DetailsPage from "./DetailsPage";
import { mockGame } from "../mocks/mockGame";
import TestRenderer from "react-test-renderer";
import { render } from "@testing-library/react";
import { Wrapper } from "../utils/Wrapper";
import React from "react";
const mockNavigate = jest.fn().mockReturnValue(mockGame.id);
jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useParams: () => ({ id: mockGame.id }),
  useNavigate: () => mockNavigate,
}));

const mockUseGames = {
  getOneGameById: jest.fn().mockResolvedValue(mockGame),
};

jest.mock("../hooks/useGamesApi", () => () => mockUseGames);

describe("Given a DetailsPage component", () => {
  describe("When it's instantiated", () => {
    test("Then it should match the screenshoot", async () => {
      const detailsSnapshot = TestRenderer.create(
        <Wrapper>
          <DetailsPage />
        </Wrapper>
      );

      expect(detailsSnapshot).toMatchSnapshot();
    });

    test("And it should call the useState function", () => {
      const mockUseState = jest.spyOn(React, "useState");
      render(
        <Wrapper>
          <DetailsPage />
        </Wrapper>
      );

      expect(mockUseState).toHaveBeenCalled();
    });
  });
});
