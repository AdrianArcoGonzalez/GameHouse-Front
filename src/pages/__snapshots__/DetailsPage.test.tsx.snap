// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a DetailsPage component When it's instantiated Then it should match the screenshoot 1`] = `
Array [
  <section
    className="sc-gsnTZi cUELPY"
    data-testid="section"
  >
    <img
      alt="backgroundImage"
      className="hero__image"
      height="200px"
      src="/images/detailsHeader.jpg"
      width="100vw"
    />
    <div
      className="hero-container"
    >
      <p
        className="hero-container__sentence"
      >
        “We look up at the same starts and see such diferent things.”(Jon Snow)
      </p>
    </div>
  </section>,
  <section
    className="sc-bczRLJ gOLlQd"
  >
    <div
      className="image-container"
    >
      <img
        alt=""
        className="details__image"
        height={320}
        src=""
        width={250}
      />
    </div>
    <article
      className="details__info-container"
    >
      <h3
        className="details__info-element-title"
      >
         
      </h3>
      <p
        className="details__info-element"
      >
         
      </p>
      <span
        className="details__info-element"
      >
        <span
          className="details__info-element--title"
        >
          Category: 
        </span>
      </span>
      <span
        className="details__info-element"
      >
        <span
          className="details__info-element--title"
        >
          Company: 
        </span>
      </span>
      <span
        className="details__info-element--owner"
      >
        <span
          className="details__info-element--title"
        >
          Created by: 
        </span>
      </span>
      <button
        className="button-delete"
        onClick={[Function]}
      >
        Delete
      </button>
    </article>
  </section>,
]
`;
